x/model/keeper/msg_server_model.go

```diff
@@ -195,5 +200,31 @@ func (k msgServer) DeleteModel(goCtx context.Context, msg *types.MsgDeleteModel)
 	// remove product from VendorProducts
 	k.RemoveVendorProduct(ctx, msg.Vid, msg.Pid)
 
+	modelVersions, err := k.ModelVersions(goCtx, &types.QueryGetModelVersionsRequest{
+		Vid: msg.Vid, 
+		Pid: msg.Pid,
+	})
+
+	if err != nil && status.Code(err) == codes.InvalidArgument {
```

Comment:
if the approach mentioned above is used, the return value `found` can be used here